### PR TITLE
Add Delete Account Option for non-root user

### LINKED ISSUE #3459: Users should be allowed to delete their account themselves
Allow users to run `ufarewell` on themselves, check to make sure it’s restricted to root otherwise.

### PR DESCRIPTION
**Pull request by yuqijin**
### Reasons for making this change

The user can use `ufarewell` cmd or the `Delete your account` button on the account page to delete his own account if the user does not own any bundles, worksheets, or groups.

Provide a delete account dialog to avoid unintentional deletion.

### Related issues

fixes #3459

### Screenshots

ezgif com-gif-maker (5)

### Checklist

* [ ] I've added a screenshot of the changes, if this is a frontend change
* [ ] I've added and/or updated tests, if this is a backend change
* [ ] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [ ] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

**Comment by adiprerepa** [comment]
why did the gif show an error?

**Comment by yuqijin** [comment]
> why did the gif show an error?

@adiprerepa Because the user owns bundles/worksheets/groups, so the user cannot delete the account.

File: codalab/rest/users.py [hunk 1]
```diff
@@ -101,8 +101,11 @@ def delete_user():
 
     request_user_id = request.user.user_id
 
-    if request_user_id != local.model.root_user_id:
-        abort(http.client.UNAUTHORIZED, 'Only root user can delete other users.')
+    if not (
+        request_user_id == local.model.root_user_id
+        or (len(user_ids) == 1 and request_user_id == user_ids[0])
```
Comment (epicfaace) [inline codalab/rest/users.py]:
```suggestion
        or (user_ids == [request_user_id])
```

File: codalab/rest/users.py [hunk 2]
```diff
@@ -135,7 +138,7 @@ def delete_user():
 
         groups = local.model.batch_get_groups(owner_id=user_id)
         if groups is not None and len(groups) > 0:
-            group_uuids = [group.uuid for group in groups]
+            group_uuids = [group['uuid'] for group in groups]
```
Comment (epicfaace) [inline codalab/rest/users.py:138]: Why this change? Was this a bug earlier?

File: frontend/src/components/UserInfo.js [hunk 3]
```diff
@@ -342,4 +353,83 @@ class AccountProfileField extends React.Component {
...
+            .catch((error) => alert(error));
+    };
+
+    return (
+        <div>
+            <Divider style={{ marginBottom: '30px' }} />
+            <h2 style={{ color: '#bc3638' }}>Delete account</h2>
+            <p>
+                To be safe, you can only delete your account if you do not own any bundles,
+                worksheets, or groups.
```
Comment (epicfaace) [inline frontend/src/components/UserInfo.js:390]: what does owning a group mean? Does this just mean being part of a group?

File: codalab/rest/users.py (hunk 2 above)
Comment (yuqijin) [reply codalab/rest/users.py:138]: yes, some bug existing due to the object/dict difference

File: frontend/src/components/UserInfo.js (hunk 3 above)
Comment (yuqijin) [reply frontend/src/components/UserInfo.js:390]: A group only has one owner, and owning a group means that the user creates the group

**Review by epicfaace — APPROVED** [review]
